### PR TITLE
Metaos rename xdai chain to gnosis chain 1170

### PR DESCRIPTION
**Pull request by HHH-GH**
## Overview

**What features/fixes does this PR include?**
Replaces occurrences of names/logos/images/icons of "xdai chain" with "gnosis chain", since xDai Chain has been renamed to Gnosis Chain.

**Please provide the GitHub issue number**
#1170 

## Implementation

The work done so far just replaces the actual icon and text that are displayed on the front end.

e.g. 
`daoHausXdaiGraphqlURL` in [`packages/backend/src/config.ts`](https://github.com/MetaFam/TheGame/blob/306f7f0dd600d42ced90841f05c548000ba1093f/packages/backend/src/config.ts#L61-L64) has not (yet?) been changed to `daoHausGnosisGraphqlURL` [as suggested by @dysbulic](https://github.com/MetaFam/TheGame/issues/1170#issuecomment-1223751604)

and in `packages/web/components/Guild/GuildForm.tsx` it's still
```
<option key="xdai" value="xdai">
    Gnosis
</option>
```
instead of
```
<option key="gnosis" value="gnosis">
    Gnosis
</option>
```
**Also**

I couldn't figure out how I could view the guild form page. Is it behind an admin login somewhere?

## Assets
Here's a screenshot of the Gnosis icon in action on a player page

image

vs. the xDAI icon

image

### CODE REVIEW

File: packages/design-system/src/icons/ChainIcon.tsx [hunk 1]
```diff
@@ -3,8 +3,9 @@ import { Tooltip } from '@chakra-ui/react';
 import React from 'react';
 
 import { EthereumIcon } from './EthereumIcon';
+// import { XDaiIcon } from './XDaiIcon'; // Replaced by GnosisIcon
```
Comment (dysbulic) [inline packages/design-system/src/icons/ChainIcon.tsx:6]:
```suggestion
```

File: packages/design-system/src/icons/ChainIcon.tsx [hunk 2]
```diff
@@ -13,8 +14,13 @@ type Props = {
 export const ChainIcon: React.FC<Props & IconProps> = ({ chain, ...props }) => {
   const lower = chain?.toLowerCase();
   const info = (() => {
+    // Figure out this part later
+    // if (lower?.includes('gnosis')) {
+    //  return { Icon: GnosisIcon, name: 'Gnosis' };
+    // }
```
Comment (dysbulic) [inline packages/design-system/src/icons/ChainIcon.tsx:20]: If you're not going to switch the value to `gnosis`, you can remove this.

```suggestion
```

File: packages/design-system/src/icons/ChainIcon.tsx [hunk 3]
```diff
@@ -13,8 +14,13 @@ type Props = {
...
     if (lower?.includes('xdai')) {
-      return { Icon: XDaiIcon, name: 'xDAI' };
+      // return { Icon: XDaiIcon, name: 'xDAI' }; // Just replace it with the Gnosis icon
+      return { Icon: GnosisIcon, name: 'Gnosis' }; // testing to see if this is enough to do it
```
Comment (dysbulic) [inline packages/design-system/src/icons/ChainIcon.tsx:23]:
```suggestion
      return { Icon: GnosisIcon, name: 'Gnosis' };
```

File: packages/design-system/src/icons/index.ts [hunk 4]
```diff
@@ -9,4 +9,5 @@ export { MeetWithWalletIcon } from './MeetWithWalletIcon';
 export { MetaMaskIcon } from './MetaMaskIcon';
 export { PolygonIcon } from './PolygonIcon';
 export { Profile } from './Profile';
-export { XDaiIcon } from './XDaiIcon';
+// export { XDaiIcon } from './XDaiIcon'; // Replaced by Gnosis icon
```
Comment (dysbulic) [inline packages/design-system/src/icons/index.ts:12]:
```suggestion
```

**Review by dysbulic — APPROVED** [review]
LGTM other than some extraneous comments that don't really do much but take up space.

**Comment by dysbulic** [comment]
> I couldn't figure out how I could view the guild form page. Is it behind an admin login somewhere?

To figure this out, first `git grep GuildForm` which outputs:
```
packages/web/components/Guild/GuildForm.tsx:64:export interface EditGuildFormInputs {
packages/web/components/Guild/GuildForm.tsx:91:): EditGuildFormInputs => {
⋮
packages/web/pages/guild/[guildname]/edit.tsx:103:          <GuildForm
⋮
packages/web/pages/join/guild/[guildname].tsx:100:          <GuildForm
```

That second path is where the form is actually displayed: `packages/web/pages/guild/[guildname]/edit.tsx`. If you're familiar with Next.js you'll recognize `[guildname]` is a placeholder to be replaced with a value. So, if you go to [`metagame.wtf/guilds/`](//metagame.wtf/guilds/) and click on a guild, you'll see a URL like: [`metagame.wtf/guild/Giveth`](//metagame.wtf/guild/Giveth). Just add [`/edit`](//metagame.wtf/guild/Giveth/edit) to the end to see the form.

**Comment by dysbulic** [comment]
Aaand, I just noticed this is a draft. Sorry if I was premature. 😸

**Comment by HHH-GH** [comment]
> Aaand, I just noticed this is a draft. Sorry if I was premature. 😸

I actually did the PR because I was looking to gather some opinions and thought having a test deployment to look at might be helpful :)

As of now, I've done the bare minimum to solve the issue. 

If that's enough, then I'd delete the extra comments and call it done.

If we want to go a bit deeper on it, then I'm thinking that I'd need some help.

e.g.
Switching
```
<option key="xdai" value="xdai">
    Gnosis
</option>
```
to
```
<option key="gnosis" value="gnosis">
    Gnosis
</option>
```
Would all the `xdai` keys in the database need to be replaced with `gnosis`?

This sort of update is also referred to in https://github.com/MetaFam/TheGame/pull/1163#discussion_r815834853

**Comment by HHH-GH** [comment]
> > I couldn't figure out how I could view the guild form page. Is it behind an admin login somewhere?
> 
> To figure this out, first `git grep GuildForm` which outputs:
> 
> ```
> packages/web/components/Guild/GuildForm.tsx:64:export interface EditGuildFormInputs {
> packages/web/components/Guild/GuildForm.tsx:91:): EditGuildFormInputs => {
> ⋮
> packages/web/pages/guild/[guildname]/edit.tsx:103:          <GuildForm
> ⋮
> packages/web/pages/join/guild/[guildname].tsx:100:          <GuildForm
> ```
> 
> That second path is where the form is actually displayed: `packages/web/pages/guild/[guildname]/edit.tsx`. If you're familiar with Next.js you'll recognize `[guildname]` is a placeholder to be replaced with a value. So, if you go to [`metagame.wtf/guilds/`](//metagame.wtf/guilds/) and click on a guild, you'll see a URL like: [`metagame.wtf/guild/Giveth`](//metagame.wtf/guild/Giveth). Just add [`/edit`](//metagame.wtf/guild/Giveth/edit) to the end to see the form.

Thanks, it's working as expected

image

**Comment by luxumbra** [comment]
Great to see more PRs from you @HHH-GH 💪🏻

**Comment by HHH-GH** [comment]
Here's my idea to finish this off.

Either 
a) Bare minimum - I'll delete the extra comments and what-not, done
b) Add more Gnosis, patch the database, clear out old xDAI

More on option b)

- All the xDAI/Gnosis Chain guilds currently in the database are identified with the `xdai` value.
- After this PR is merged, any new xDAI/Gnosis Chain guilds will be identified with the `gnosis` value. (see `GuildForm.tsx` below)
- Until the database is patched to change all the previous `xdai` values to the new `gnosis` value, the Gnosis icon will be displayed for both `xdai` and `gnosis` values. (See `ChainIcon.tsx` below)
- After the database is patched (if?), the extra code in `ChainIcon.tsx` can be removed.

Implemented as follows

In `GuildForm.tsx`:
```
<option key="gnosis" value="gnosis">
    Gnosis
</option>
```

In `ChainIcon.tsx`
```
import { EthereumIcon } from './EthereumIcon';
import { GnosisIcon } from './GnosisIcon';
import { PolygonIcon } from './PolygonIcon';

type Props = {
  chain?: string;
};

export const ChainIcon: React.FC<Props & IconProps> = ({ chain, ...props }) => {
  const lower = chain?.toLowerCase();
  const info = (() => {
    if (lower?.includes('gnosis')) {
        return { Icon: GnosisIcon, name: 'Gnosis' };
    }
    if (lower?.includes('xdai')) {
      return { Icon: GnosisIcon, name: 'Gnosis' }; // Legacy code: Prior to 2022/09 an xDai icon was used, now replaced with Gnosis (but in the database it's still xdai)
    }
    if (lower?.includes('polygon')) {
      return { Icon: PolygonIcon, name: 'Polygon' };
    }
    return { Icon: EthereumIcon, name: 'Mainnet' };
  })();
  return (
    <Tooltip label={`on ${info.name}`} hasArrow>
      <info.Icon {...props} />
    </Tooltip>
  );
};
```

Option b) *sort of* doesn't actually finish this off because there's still the database patch and subsequent updates to do. If someone points me to the database structure I can write SQL to patch it. But I'm not sure if I'd be the best person to do the actual patching.

?

**Comment by HHH-GH** [comment]
Have implemented plan b, with some possibly over verbose commenting.
Ready for review @dysbulic

File: packages/design-system/src/icons/ChainIcon.tsx [hunk 5]
```diff
@@ -13,8 +13,18 @@ type Props = {
 export const ChainIcon: React.FC<Props & IconProps> = ({ chain, ...props }) => {
   const lower = chain?.toLowerCase();
   const info = (() => {
+    if (lower?.includes('gnosis')) {
+      return { Icon: GnosisIcon, name: 'Gnosis' };
+    }
     if (lower?.includes('xdai')) {
```
Comment (dysbulic) [inline packages/design-system/src/icons/ChainIcon.tsx:19]:
```suggestion
    if (lower?.includes('gnosis') || lower?.includes('xdai')) {
```

**Review by dysbulic — APPROVED** [review]
There's one place where I would have written things differently *(as an "or" rather than two separate "`if`"s)*, but it looks good to me.

I'm assuming you've tested everything to your satisfaction, so if you're satisfied, so am I.

**Review by alalonde — APPROVED** [review]

**Comment by HHH-GH** [comment]
> There's one place where I would have written things differently _(as an "or" rather than two separate "`if`"s)_, but it looks good to me.

Good point 👍

thx for the approve @alalonde

**Pull request merged**